pennylane/operation.py

```diff
@@ -1033,8 +1033,17 @@ def _check_batching(self, params):
         """
         self._batch_size = None
 
+        if len(params) == 0:
+            return
+
+        # discriminate between numpy_like `params` to accelerate two lines below
+        numpy_like = all(hasattr(p, "ndim") and hasattr(p, "shape") for p in params)
+
         try:
-            ndims = tuple(qml.math.ndim(p) for p in params)
+            if numpy_like:
+                ndims = tuple(p.ndim for p in params)
+            else:
+                ndims = tuple(qml.math.ndim(p) for p in params)
```

Comment:
@albi3ro Would you like me to remove these changes before merging?

Reply:
Yes, this should probably go in a separate PR.